src/deepsparse/engine.py

```diff
@@ -117,6 +118,18 @@ def _validate_num_cores(num_cores: Union[None, int]) -> int:
 
     return num_cores
 
+def _validate_num_streams(num_streams: Union[None, int], num_cores: int) -> int:
+    if not num_streams:
+        num_streams = NUM_STREAMS
+
+    if num_streams < 0:
+        raise ValueError("num_streams must be greater than 0")
+
+    if num_streams > num_cores:
+        raise ValueError("num_cores cannot exceed num_cores.")
```

Comment:
```suggestion
        raise ValueError("num_streams cannot exceed num_cores.")
```

Reply:
Derp. I also reformatted. Trying to remember how to push after rebasing...